airflow/ui/src/components/TriggerDag/TriggerDAGTextButton.tsx

```diff
@@ -38,13 +38,15 @@ const TriggerDAGIconButton: React.FC<Props> = ({ dag }) => {
         Trigger
       </Button>
 
-      <TriggerDAGModal
-        dagDisplayName={dag.dag_display_name}
-        dagId={dag.dag_id}
-        isPaused={dag.is_paused}
-        onClose={onClose}
-        open={open}
-      />
+      {open ? (
```

Comment:
Why are we wrapping the modal here?

Reply:
TriggerDAGModal have a dag details api call, which will be even called when the button is not clicked, by wrapping it I am not rendering until opened, not wrapping it will have ,multiple calls when on dag list page.

Reply:
I think `lazyMount` and `unmountOnExit` should handle this. We can also add `{enabled: open}` to the get dag details request